### ISSUE TITLE
Just curious about your thought process when deciding to write this

### ISSUE THREAD

**Issue by macgyver** [matankb/fetch-reject#4]
I admit I was also a bit confused when I realized that a 4xx or 5xx response was a "resolved" promise since this differed from the behavior popularized by jQuery, but I assumed that the WHATWG had made a considered decision to standardize this way. Did you consult with any of the writers of the fetch spec to see what their reasoning was?

**Comment by matankb** [comment]
First off - this project was initially used in a fetch-heavy personal project and wasn't originally intended to be used by others. I made it into its own modules because I noticed that I was writing the same lines over and over again. Anyway:

My thought process consisted of two things: a) intent vs. specification and b) separation of concerns

a) The original fetch spec does make sense, if you think of it in a certain way. The idea is that you ask for a response with `fetch` and you get one. If you don't get any response, then it'll throw. This follows the specification closely, and is solely concerned with the actual `fetch`-ing, but not how the request went. However, I prefer to think on a higher level - when I call `fetch`, especially for non-GET requests, my actual intent is to _do_ something, from POST-ing a new item to the DB to updating an existing document. Therefore, if the server returns a non-200 response, it should be treated as an error and handled as such.

b) I noticed how, without this module, I was still handling 4xx and 5xx responses, just in `.then` handlers.  By throwing an error, you're able to catch it where it should be handled - in `.catch`.

So that's my reasoning. As I said, this modules was originally meant to be used only by myself - this line of thought works for some people and doesn't for others. I hope this helps!
